README.md
<img alt="React Native Basic Modal" src="assets/logo.png" width="1050"/>

[![Battle Tested ✅](https://img.shields.io/badge/-Battle--Tested%20%E2%9C%85-03666e?style=for-the-badge)](https://github.com/WrathChaos/react-native-basic-modal)

[![Basic & Elegant UI Modal for React Native](https://img.shields.io/badge/-Basic%20%26%20Elegant%20UI%20Modal%20for%20React%20Native-lightgrey?style=for-the-badge)](https://github.com/WrathChaos/react-native-basic-modal)

[![npm version](https://img.shields.io/npm/v/react-native-basic-modal.svg?style=for-the-badge)](https://www.npmjs.com/package/react-native-basic-modal)
[![npm](https://img.shields.io/npm/dt/react-native-basic-modal.svg?style=for-the-badge)](https://www.npmjs.com/package/react-native-basic-modal)
![Platform - Android and iOS](https://img.shields.io/badge/platform-Android%20%7C%20iOS-blue.svg?style=for-the-badge)
[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg?style=for-the-badge)](https://opensource.org/licenses/MIT)
[![styled with prettier](https://img.shields.io/badge/styled_with-prettier-ff69b4.svg?style=for-the-badge)](https://github.com/prettier/prettier)

<p align="center">
  <img alt="React Native Halloween Modal"
        src="assets/Screenshots/halloween.png" width="49%" height="710"/>
  <img alt="React Native Basic Modal"
        src="assets/Screenshots/example.png" width="49%" />
</p>

# Installation

Add the dependency:

```ruby
npm i react-native-basic-modal
```

## Peer Dependencies

###### IMPORTANT! You need install them

```js
"react": ">= 16.x.x",
"react-native": ">= 0.55.x",
"react-native-modal": ">= 11.4.0"
```

# Usage

## Component Options

* BasicModal
* HalloweenModal

## Import

```js
import { BasicModal, HalloweenModal } from "react-native-basic-modal";
```
## BasicModal Usage

### Default Usage

```js
<BasicModal isVisible />
```


### Customizable Usage

```js
<BasicModal 
  isVisible
  title="Warning!" 
  desciption="Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book."
  firstButtonOnPress={()=> {}}  
  secondButtonOnPress={()=> {}}  
/>
```

## HalloweenModal

### Important note: You need to add "halloweenTheme" prop!

```js
<HalloweenModal halloweenTheme isVisible />
```

You can use the same customization options as BasicModal. HalloweenModal just a special theme for Halloween :)


### How can I hide the modal by pressing outside of its content?

The prop `onBackdropPress` allows you to handle this situation:

```javascript
<BasicModal
  isVisible={this.state.isVisible}
  onBackdropPress={() => this.setState({ isVisible: false })}
/>
```

# For more FAQ about Modal: 

#### [Modal FAQ](https://github.com/react-native-community/react-native-modal#frequently-asked-questions)

# Configuration - Props

| Property                    |   Type    |  Default  | Description                                  |
| --------------------------- | :-------: | :-------: | -------------------------------------------- |
| width                       |  number   |    90%    | change the modal's width                     |
| height                      |  number   |    185    | change the modal's height                    |
| isVisible                   |   bool    |   false   | set the modal's visibility                   |
| title                       |  string   | Hold on!  | set your own title text                      |
| description                 |  string   | too long  | set your own description text                |
| titleColor                  |   color   |  #212121  | change the title's text color                |
| descColor                   |   color   |  #b5b5b5  | change the description's text color          |
| titleStyle                  |   style   |   style   | set your own style for title text            |
| descStyle                   |   style   |   style   | set your own style for description text      |
| backgroundColor             |   color   |   #fff    | change the modal's background color          |
| firstButtonText             |  string   |   SKIP    | set your own button text                     |
| secondButtonText            |   color   |   NEXT    | set your own button text                     |
| firstButtonTextColor        |   color   |   #fff    | change the first button's text color         |
| secondButtonTextColor       |   color   |   #fff    | change the second button's text color        |
| firstButtonBackgroundColor  |   color   |  #b5b6cf  | change the first button's background color   |
| secondButtonBackgroundColor |   color   |  #895aaf  | change the second button's background color  |
| firstButtonOnPress          | function  | undefined | set your own onPress function                |
| secondButtonOnPress         | function  | undefined | set your own onPress function                |
| firstButtonComponent        | component |  default  | set your own button component for first one  |
| secondButtonComponent       | component |  default  | set your own button component for second one |


## Credits

Thank you so much for this awesome theme assets for **[Adem 'Allecroom' Kotan](https://dribbble.com/Allecroom)**
You can hire him for 2D, 3D assets :) 

### **[Hire him!](mailto:alperademkotan@gmail.com)**

Also, I inspired by [Orizon Design](https://www.instagram.com/p/BtHuW3sBpkG/) Thank you so much guys, nice UI / UX :)

## Future Plans

- [x] ~~LICENSE~~
- [x] ~~Halloween Theme~~
- [ ] Write an article about the lib on Medium

# Change Log

## [0.0.5](https://github.com/WrathChaos/react-native-basic-modal/tree/0.0.5) (2019-10-13)
[Full Changelog](https://github.com/WrathChaos/react-native-basic-modal/compare/0.0.1...0.0.5)

## [0.0.1](https://github.com/WrathChaos/react-native-basic-modal/tree/0.0.1) (2019-10-13)
[Full Changelog](https://github.com/WrathChaos/react-native-basic-modal/compare/0.0.2...0.0.1)

\* *This Change Log was automatically generated by [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator)*

## Author

FreakyCoder, kurayogun@gmail.com

## License

React Native Basic Modal is available under the MIT license. See the LICENSE file for more info.


CHANGELOG.md
# Change Log

## [0.0.5](https://github.com/WrathChaos/react-native-basic-modal/tree/0.0.5) (2019-10-13)
[Full Changelog](https://github.com/WrathChaos/react-native-basic-modal/compare/0.0.1...0.0.5)

## [0.0.1](https://github.com/WrathChaos/react-native-basic-modal/tree/0.0.1) (2019-10-13)
[Full Changelog](https://github.com/WrathChaos/react-native-basic-modal/compare/0.0.2...0.0.1)

\* *This Change Log was automatically generated by [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator)*

lib/src/BasicModal/BasicModal.js
import React from "react";
import PropTypes from "prop-types";
import { Text, View, Image } from "react-native";
import Modal from "react-native-modal";
import Button from "../components/Button/Button";
import LinearGradient from "react-native-linear-gradient";
import styles, { container, _titleStyle, _descStyle } from "./BasicModal.style";

const BasicModal = props => {
  renderButtonContainer = () => {
    const {
      secondButtonText,
      firstButtonOnPress,
      secondButtonOnPress,
      firstButtonTextColor,
      firstButtonComponent,
      secondButtonTextColor,
      secondButtonComponent,
      firstButtonBackgroundColor,
      secondButtonBackgroundColor
    } = props;
    return (
      <View style={styles.buttonContainer}>
        <View style={styles.firstButtonContainer}>
          <Button
            buttonText={firstButtonText}
            onPress={firstButtonOnPress}
            buttonComponent={firstButtonComponent}
            buttonTextColor={firstButtonTextColor}
            backgroundColor={firstButtonBackgroundColor}
          />
        </View>
        <Button
          buttonText={secondButtonText}
          onPress={secondButtonOnPress}
          buttonComponent={secondButtonComponent}
          buttonTextColor={secondButtonTextColor}
          backgroundColor={secondButtonBackgroundColor}
        />
      </View>
    );
  };

  renderTitleContainer = () => {
    const {
      title,
      descColor,
      descStyle,
      titleStyle,
      titleColor,
      description
    } = props;
    return (
      <View style={styles.containerGlue}>
        <Text style={titleStyle || _titleStyle(titleColor)}>{title}</Text>
        <Text style={descStyle || _descStyle(descColor)}>{description}</Text>
      </View>
    );
  };

  renderHalloweenModal = () => {
    const { height, width, backgroundColor, children } = props;
    return (
      <View style={container(height, width, backgroundColor)}>
        {children}
        <LinearGradient
          end={{ x: 1, y: 0 }}
          start={{ x: 0, y: 0 }}
          colors={styles.gradientColors}
          style={styles.halloweenGradientContainer}
        >
          <Image
            resizeMode="contain"
            style={styles.birdyImageStyle}
            source={require("../fundamental-assets/birdy.png")}
          />
          {renderTitleContainer()}
          {renderButtonContainer()}
        </LinearGradient>
      </View>
    );
  };

  renderNormalModal = () => {
    const { height, width, backgroundColor, children } = props;
    return (
      <View style={container(height, width, backgroundColor)}>
        {children}
        {renderTitleContainer()}
        {renderButtonContainer()}
      </View>
    );
  };

  const { isVisible, halloweenTheme, firstButtonText, onBackdropPress } = props;

  return (
    <Modal
      isVisible={isVisible}
      style={styles.modalContainer}
      onBackdropPress={onBackdropPress}
    >
      {halloweenTheme ? renderHalloweenModal() : renderNormalModal()}
    </Modal>
  );
};

BasicModal.propTypes = {
  title: PropTypes.string,
  isVisible: PropTypes.bool,
  descColor: PropTypes.string,
  titleColor: PropTypes.string,
  description: PropTypes.string,
  halloweenTheme: PropTypes.bool,
  backgroundColor: PropTypes.string,
  firstButtonText: PropTypes.string,
  secondButtonText: PropTypes.string,
  firstButtonTextColor: PropTypes.string,
  secondButtonTextColor: PropTypes.string,
  firstButtonBackgroundColor: PropTypes.string,
  secondButtonBackgroundColor: PropTypes.string
};

BasicModal.defaultProps = {
  width: "90%",
  height: 185,
  isVisible: false,
  title: "Hold on!",
  descColor: "#b5b5b5",
  titleColor: "#212121",
  backgroundColor: "#fff",
  firstButtonText: "SKIP",
  halloweenTheme: false,
  secondButtonText: "NEXT",
  firstButtonTextColor: "#fff",
  secondButtonTextColor: "#fff",
  firstButtonBackgroundColor: "#b5b6cf",
  secondButtonBackgroundColor: "#895aaf",
  description:
    "Would you like to try our short tutorial to show you how this app work? It will take no more than 30 seconds!"
};

export default BasicModal;
